diff --git a/node_modules/@eslint/eslintrc/lib/shared/config-validator.js b/node_modules/@eslint/eslintrc/lib/shared/config-validator.js
index 6857e7a..3f8635c 100644
--- a/node_modules/@eslint/eslintrc/lib/shared/config-validator.js
+++ b/node_modules/@eslint/eslintrc/lib/shared/config-validator.js
@@ -324,12 +324,6 @@ export default class ConfigValidator {
      * @throws {Error} If the config is invalid.
      */
     validateConfigSchema(config, source = null) {
-        validateSchema = validateSchema || ajv.compile(configSchema);
-
-        if (!validateSchema(config)) {
-            throw new Error(`ESLint configuration in ${source} is invalid:\n${this.formatErrors(validateSchema.errors)}`);
-        }
-
         if (Object.hasOwn(config, "ecmaFeatures")) {
             emitDeprecationWarning(source, "ESLINT_LEGACY_ECMAFEATURES");
         }
