diff --git a/node_modules/bob-the-bundler/dist/commands/build.js b/node_modules/bob-the-bundler/dist/commands/build.js
index e690b91..bbd031c 100644
--- a/node_modules/bob-the-bundler/dist/commands/build.js
+++ b/node_modules/bob-the-bundler/dist/commands/build.js
@@ -66,7 +66,7 @@ async function buildTypeScript(buildPath, options) {
     ]));
     assertTypeScriptBuildResult(await execa('npx', [
         'tsc',
-        ...(options.tsconfig ? ['--project', options.tsconfig] : []),
+        ...(tsconfig ? ['--project', tsconfig] : []),
         ...compilerOptionsToArgs(typeScriptCompilerOptions('cjs')),
         ...(options.incremental ? ['--incremental'] : []),
         '--outDir',
